### PR TITLE
Game options visitor

### PR DESCRIPTION
**Pull request by liach**
so they don't write a code twice for save and load

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by jaskarth — APPROVED** [review]

File: mappings/net/minecraft/client/option/GameOptions.mapping [hunk 1]
```diff
@@ -139,10 +140,65 @@ CLASS net/minecraft/class_315 net/minecraft/client/option/GameOptions
...
+		ARG 0 value
+	METHOD method_33675 loadArm (Ljava/lang/String;)Lnet/minecraft/class_1306;
+		ARG 0 arm
+	CLASS 2
+		METHOD method_33676 find (Ljava/lang/String;)Ljava/lang/String;
+			ARG 1 key
+	CLASS 3
+		METHOD method_33677 print (Ljava/lang/String;)V
+			ARG 1 key
+	CLASS class_5823 Visitor
```
Comment (i509VCB) [inline mappings/net/minecraft/client/option/GameOptions.mapping:179]: Hmm I wonder, would this allow injection of custom options.

Comment (liach) [reply mappings/net/minecraft/client/option/GameOptions.mapping:179]: Yes if you can bytecode hack `accept` to add an extra call to account for another option.
Unfortunately afaik, mixin still doesn't work nicely with these package classes, right? Kaboom time as this interface indeed is package private https://maven.fabricmc.net/docs/yarn-21w05a+build.1/net/minecraft/client/option/GameOptions.class_5823.html

Comment (Juuxel) [reply mappings/net/minecraft/client/option/GameOptions.mapping:179]: You can inject to them just fine but mixin doesn't really help with extending/implementing/creating them.

Comment (liach) [reply mappings/net/minecraft/client/option/GameOptions.mapping:179]: Well, mixin has trouble calling methods from this interface, though you most likely don't need new methods in this interface that handle extra data types

**Review by Juuxel — APPROVED** [review]

**Pull request merged**